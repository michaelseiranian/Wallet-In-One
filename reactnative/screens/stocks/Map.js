import MapView from 'react-native-maps';
import { Marker } from 'react-native-maps';
import { StyleSheet } from 'react-native';

/**
 * Component that takes in a latitude and longitude and displays a map with a marker at the specified location
 */
export default function Map({latitude, longitude}){
  if(latitude && longitude){
    return(
        <MapView
        style={{...StyleSheet.absoluteFillObject}}
        region={{
          latitude: latitude,
          longitude: longitude,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
        }}
        testID={"mapView"}
      >
    <Marker coordinate={{ latitude: latitude, longitude: longitude }} testID={"marker"} />
      </MapView>
    )
  }
  else{
    return null;
  }
}