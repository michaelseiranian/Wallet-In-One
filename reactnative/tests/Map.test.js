import React from 'react';
import Map from '../screens/stocks/Map';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';

describe('<Map />', () => {
    it('map test regular values', () => {
        const map = renderer.create(<Map latitude={10} longitude={15}/>);
        expect(map).toMatchSnapshot();
      });

    it('map test marker', () => {
        const { getByTestId } = render(<Map latitude={37} longitude={50} />);
        expect(getByTestId("marker")).toBeDefined();
        expect((getByTestId("marker").props.coordinate.latitude)).toBe(37);
        expect((getByTestId("marker").props.coordinate.longitude)).toBe(50);
      });

    it('map test mapView', () => {
        const { getByTestId } = render(<Map latitude={37} longitude={50} />);
        expect((getByTestId("mapView").props.region.latitude)).toBe(37);
        expect((getByTestId("mapView").props.region.longitude)).toBe(50);
        expect((getByTestId("mapView").props.style.position)).toBe('absolute');
        expect(getByTestId("mapView")).toBeDefined();
      });

    it('map test negative values', () => {
        const map = renderer.create(<Map latitude={-10} longitude={-15}/>);
        expect(map).toMatchSnapshot();
      });
      
    it('map test zero values', () => {
        const map = renderer.create(<Map latitude={0} longitude={0}/>);
        expect(map).toMatchSnapshot();
      }); 
      
    it('map test null values returns null', () => {
        const map = renderer.create(<Map latitude={null} longitude={null}/>).toJSON();
        expect(map).toBeNull();
        expect(map).toMatchSnapshot();
      });
      
    it('map test no values returns null', () => {
        const map = renderer.create(<Map />).toJSON();
        expect(map).toBeNull();
        expect(map).toMatchSnapshot();
      });      
  });



