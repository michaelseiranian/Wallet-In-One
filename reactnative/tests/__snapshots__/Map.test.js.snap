// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> map test negative values 1`] = `
<AIRMap
  initialRegion={null}
  onChange={[Function]}
  onMapReady={[Function]}
  region={
    {
      "latitude": -10,
      "latitudeDelta": 0.0922,
      "longitude": -15,
      "longitudeDelta": 0.0421,
    }
  }
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  testID="mapView"
>
  <AIRMapMarker
    coordinate={
      {
        "latitude": -10,
        "longitude": -15,
      }
    }
    onPress={[Function]}
    style={
      [
        {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    testID="marker"
  />
</AIRMap>
`;

exports[`<Map /> map test no values returns null 1`] = `null`;

exports[`<Map /> map test null values returns null 1`] = `null`;

exports[`<Map /> map test regular values 1`] = `
<AIRMap
  initialRegion={null}
  onChange={[Function]}
  onMapReady={[Function]}
  region={
    {
      "latitude": 10,
      "latitudeDelta": 0.0922,
      "longitude": 15,
      "longitudeDelta": 0.0421,
    }
  }
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  testID="mapView"
>
  <AIRMapMarker
    coordinate={
      {
        "latitude": 10,
        "longitude": 15,
      }
    }
    onPress={[Function]}
    style={
      [
        {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    testID="marker"
  />
</AIRMap>
`;

exports[`<Map /> map test zero values 1`] = `null`;
